src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
-// write code here
+const list = document.querySelectorAll('li');
+
+for (const title of list) {
+  if (title.firstChild && title.firstChild.nodeType === 3) {
+    const element = document.createElement('span');
+
+    element.textContent = title.firstChild.textContent;
+    title.firstChild.replaceWith(element);
+  }
+}
+
+document.addEventListener('click', (e) => {
+  const element = e.target.closest('span');
+  const liElement = element.closest('li');
```

Comment:
The use of `closest('span')` and `closest('li')` assumes that the clicked element is always within a `span` and `li`. If the click occurs outside these elements, `element` or `liElement` could be `null`, leading to potential errors. Consider adding a null check for `element` before calling `closest('li')`.